src/dcd/server/autocomplete/util.d

```diff
@@ -755,14 +755,120 @@ unittest
 
 AutocompleteResponse.Completion makeSymbolCompletionInfo(const DSymbol* symbol, char kind)
 {
-	string definition;
-	if ((kind == CompletionKind.variableName || kind == CompletionKind.memberVariableName) && symbol.type)
-		definition = symbol.type.name ~ ' ' ~ symbol.name;
-	else if (kind == CompletionKind.enumMember)
-		definition = symbol.name; // TODO: add enum value to definition string
+	string definition = symbol.callTip;
+	string symbolName = symbol.name;
+	char completionKind = kind;
+
+	// if symbol has a type, then let's try to resolve it
+	// and properly format the definition
+	if (symbol.type)
+	{
+		DSymbol* sym = cast(DSymbol*) symbol;
+		DSymbol* symType =  sym.type;
+
+		while (symType && sym && sym.type != sym)
```

Comment:
this should have a counter / max limit just in case there are infinite cycles to break after e.g. 50 iterations or so